### DISCUSSION TITLE
Version Migrations

### DISCUSSION THREAD

**Discussion by mackov83** [netbox-community/netbox#5727] [Q&A] [answered]
Hi,

I am relatively new to NetBox and have inherited a 2.7.8 release. On top of this the OS much to my dismay is currently on Ubuntu 19.10 - sudo apt update won't work anymore :)

I attempted:

- An upgrade from 19.10 to 20.04.2 - worked fine
- An upgrade from NetBox 2.7.8 to 2.10.4 - seemed ok

While there were quite a few things that tripped me up, the only thing that forced my rollback was that LDAP failed to work - though was working prior, and also works after rollback - unfortunately this included the OS and NetBox.

A couple of questions:
1. in 2.7.8 I had a gunicorn_config.py, whereas doco says gunicorn.py - is this a legacy thing, or whoever deployed used non-standard names? I simply renamed the file and the services started on 2.10.4
2. Any ideas why LDAP might have had an issue - i.e. known bugs? Setting up LDAP logging as per the documentation resulted in NetBox not starting, so no help there
3. Is it possible to dump the DB, perform fresh Ubuntu 20.04 and NetBox 2.10 install and import the database? If so, any good guides available?

**Comment by candlerb** [comment]
> in 2.7.8 I had a gunicorn_config.py, whereas doco says gunicorn.py - is this a legacy thing, or whoever deployed used non-standard names? I simply renamed the file and the services started on 2.10.4

That's not a big deal, just use whatever your systemd unit file points at.

Re LDAP: I don't know, you could also ask on the google group where there has been some LDAP discussion.  If Netbox doesn't start, then you should be able to show the error message and that might clarify the problem.

> Is it possible to dump the DB, perform fresh Ubuntu 20.04 and NetBox 2.10 install and import the database? If so, any good guides available?

Yes it is, and the guide is here: https://netbox.readthedocs.io/en/stable/administration/replicating-netbox/

**Reply by mackov83** [reply]
Great, thanks for the info

**Comment by rodvand** [comment, marked as answer]
> 2. Any ideas why LDAP might have had an issue - i.e. known bugs? Setting up LDAP logging as per the documentation resulted in NetBox not starting, so no help there

Did you change your `configuration.py` as per the changes in 2.9.0 (https://github.com/netbox-community/netbox/releases/tag/v2.9.0)?
From the release notes: 
"If in use, LDAP authentication must be enabled by setting REMOTE_AUTH_BACKEND to 'netbox.authentication.LDAPBackend'. (LDAP configuration parameters in ldap_config.py remain unchanged.)"

**Reply by mackov83** [reply]
Thank you, I completely missed this! I will give that a go and update

**Comment by mackov83** [comment]
Hi @rodvand, The changes that were included in 2.7.9 were indeed the fix in this case. We are now upgraded from Ubuntu 19.10 and NetBox 2.7.9 to Ubuntu 20.04.2 and NetBox 2.10.4
